Started implementing window vibrancy

lewis-carson opened the pull request:
Issue #300 proposes adding support for changing window vibrancy/adding acrylic effect. I think this can be done with minimal effort and changes to the existing codebase. While this isn't ready to be merged by far, I think it's a good start and can be worked into a functional update with a few hours more work.

panekj commented on Cargo.toml:
```diff
@@ -44,6 +44,7 @@ floem_reactive = { path = "reactive", version = "0.1.0" }
 floem-winit = { version = "0.29.4", features = ["rwh_05"] }
 image = { version = "0.24", features = ["jpeg", "png"] }
 copypasta = { version = "0.10.0", default-features = false, features = ["wayland", "x11"] }
+floem-window-vibrancy = "0.4.3"
```
Why does it have to be separate crate?

panekj replied:
Also it would've been *nice* to ask us first about publishing crate with `floem` name

lewis-carson commented:
> Looking at the window vibrancy crate it looked like in order to apply the effect the code just needs access to the raw window handle.

I've deleted the fork; sorry for being presumptuous. I've updated the pull request to allow the user to obtain the raw window handle for their own usage. It should be functional but I've assumed that the window handle doesn't change over the lifespan of a window.

jrmoulton commented:
This looks fine to me. I don't love the name of the function in decorator. `raw_window_handle` isn't super descriptive but I can't think of anything better. 

@dzhou121

dzhou121 commented:
This approach feels a bit hacky for me. Ideally it should be one option in WindowConfig, and add the vibrancy support to our winit fork.

panekj commented:
I don't think Floem should expose `WindowHandle` at all
